src/dirwatcher.js

```diff
@@ -0,0 +1,77 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as util from 'util';
+import * as EventEmitter from 'events';
+import logger from '../config/logger';
+
+class dirwatcher extends EventEmitter {
+    constructor() {
+        super();
+        this.modifiedFiles = [];
+        this.modificationDates = {};
+    }
+
+    watch(path, delay) {   
+        const timeout = setInterval(()=> {
```

Comment:
it might make sense to return interval ID from watch function to clean up resources when needed.